lib/views/home.dart

```diff
@@ -48,3 +49,67 @@ class HomePage extends StatelessWidget {
     );
   }
 }
+
+ _checkConnectivity(context) async {
+    var result = await Connectivity().checkConnectivity();
+    if (result == ConnectivityResult.none){
+      _displayDialog(
+        context,
+          Text(
+              "No Connection",
+
+              style: TextStyle(
+                  color: Colors.white,
+                  fontSize: 14.0,
+                  fontWeight: FontWeight.bold,
+                  letterSpacing: 2.0,
+                  fontFamily: "fonts/BalsamiqSans-Bold.ttf"
+              ),
+            textAlign: TextAlign.center,
+          ),
+          Text(
+            "Oops, You do not have internet connection. \n\nClick reload to try again",
+            style: TextStyle(
+                color: Colors.white,
+                fontSize: 16.0,
+                letterSpacing: 2.0,
+                fontFamily: "fonts/BalsamiqSans-Bold.ttf"
+            ),
+            textAlign: TextAlign.center,
+          )
+      );
+    }else{
+      Navigator.pushReplacement(context,
+          MaterialPageRoute(builder: (context) => QuizPlay()));
+    }
+}
+
+void _displayDialog(context,title,text) {
+  showDialog(
+      context: context,
+    builder: (context){
+        return AlertDialog(
+          title: title,
+          content: text,
+          backgroundColor: Color(0xff071a35),
+          shape: RoundedRectangleBorder(
+            borderRadius: BorderRadius.circular(16.0)
+          ),
+          elevation: 1,
+          actions: <Widget>[
+            FlatButton.icon(
+                onPressed: (){
+                  Navigator.pushReplacement(context,
+                      MaterialPageRoute(builder: (context) => HomePage()));
```

Comment:
Bro here, calling `Navigator.pushReplacement()` will create another` HomePage()` on top of what we currently have, i.e the Dialog context. So each time I click reload a new `HomePage()` screen is created.
A better approach is to dismiss the dialog and show the home page. Somehting line `Navigator.pop(context)` , instead of `Navigator.pushReplacement()`
To test how it currently behaves, disable network connectivity in your device and run the app, click on the reload button multiple times, and then click on the phone's back button, you'll see that you have created multiple home pages, one for each click of the reload.